src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.cs

```diff
@@ -21,18 +26,33 @@ public virtual void Write(string message)
                     s_WriteCore(string.Empty);
                     return;
                 }
-                if (s_needIndent)
+                if (s_needIndent && !_ignoreIndentation)
```

Comment:
This may be easier to understand:

```
if (_ignoreIndentation)
{
    s_WriteCore(message);
    return;
}
```

Reply:
(And also saves unnecessary EndsWith call.)

Reply:
Thanks

Reply:
UPDATE:
I don't need _ignoreIndentation (refer to 2544c86) anymore. 
I can override `IndentLevel` for `DefaultTraceDebugProvider` instead.
(further explanation in https://github.com/dotnet/corefx/pull/33020/files#r227988644)